src/main/java/software/sigma/internship/service/UserService.java

```diff
@@ -0,0 +1,15 @@
+package software.sigma.internship.service;
+
+import software.sigma.internship.dto.AuthUserDto;
+import software.sigma.internship.dto.UserDto;
+import software.sigma.internship.exception.UserAlreadyExistException;
+
+public interface UserService {
+    UserDto getUserById(long id);
+
+    UserDto getUserByEmail(String email);
+
+    UserDto register(AuthUserDto user) throws UserAlreadyExistException;
```

Comment:
We should throw runtime exceptions in our services and handle them in ExceptionHandler